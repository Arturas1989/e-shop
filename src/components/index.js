export { Header } from './Header/Header';
export { NavigationMenu } from './Header/Menu/NavigationMenu/NavigationMenu';
export { LoginMenu } from './Header/Menu/LoginMenu/LoginMenu';
export { Hamburger } from './Header/Hamburger/Hamburger';
export { CloseMenu } from './Header/Menu/NavigationMenu/CloseMenu';
export { LoginLink } from './Header/Menu/LoginMenu/LoginLink';

export { SearchBar } from './SearchBar/SearchBar';
export { InputBar } from './SearchBar/InputBar';

export { Footer } from './Footer/Footer';

export { PrimaryButton } from './Buttons/PrimaryButton/PrimaryButton';
export { SecondaryButton } from './Buttons/SecondaryButton/SecondaryButton';

export { Rating } from './ProductCard/Rating/Rating';
export { BestSeller } from './ProductCard/BestSeller/BestSeller';
export { Heading } from './Heading/Heading';
