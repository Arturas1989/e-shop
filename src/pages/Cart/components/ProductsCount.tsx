import { useCartContext } from "../../../contexts/cartContext";
import { useCartProducts } from "../../../hooks";

export const ProductsCount = () => {
  const {cart} = useCartContext()!;
  const [, products] = useCartProducts(cart);

  return (
    <span>({products?.length})</span>
  )
};
