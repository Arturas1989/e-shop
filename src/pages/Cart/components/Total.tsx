import { Product } from "../../../types";

type TotalProps = {
  products: Product[] | null;
};

export const Total = ({ products }: TotalProps) => {
  let total;
  if(products)total = products.reduce((acc, product) => acc + product.price * product.quantity!, 0);
  return (
    <div className="total-row">
      <h2>Total amount:</h2>
      <span>{total}$</span>
    </div>
  )
};

