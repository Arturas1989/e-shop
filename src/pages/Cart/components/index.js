export { ProductsCount } from './ProductsCount';
export { CartCard } from './Cartcard/CartCard';
export { CartProducts } from './CartProducts';
