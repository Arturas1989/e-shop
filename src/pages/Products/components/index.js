export { FilterOpenButton } from './Filter/FilterOpenButton';
export { ProductsCount } from './ProductsCount';
export { Filter } from './Filter/Filter';
export { InputGroup } from './Filter/InputGroup';
