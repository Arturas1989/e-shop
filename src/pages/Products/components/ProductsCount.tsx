import { Loading } from "../../../components";
import { useProductsContext } from "../../../contexts/productContext";

export const ProductsCount = () => {
  const {products, isProductsLoading} = useProductsContext()!
  return (
    isProductsLoading ? <h2>All products ({products?.length})</h2> : <Loading />
  )
};
