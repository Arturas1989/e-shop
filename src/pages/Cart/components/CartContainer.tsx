import { SecondaryButton } from "../../../components";
import { Product } from "../../../types";
import { CartCard } from "./Cartcard/CartCard";
import { Total } from "./Total";

type CartContainerProps = {
  products: Product[] | null;
};

export const CartContainer = ({ products }: CartContainerProps) => {
  return (
    <>
      {products?.map((product) => (
        <CartCard key={product.id} product={product} />
      ))}
      <Total products={products} />
      <SecondaryButton>
        Place order<i className="bi bi-arrow-right"></i>
      </SecondaryButton>
    </>
  );
};
