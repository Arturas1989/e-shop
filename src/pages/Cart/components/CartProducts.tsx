import { Loading, SecondaryButton } from '../../../components';
import { useCartContext } from '../../../contexts/cartContext';
import { useCartProducts } from '../../../hooks';
import { CartCard } from './Cartcard/CartCard';
import { Total } from './Total';

type CartProductsProps = {
  name?: string;
};

export const CartProducts = ({ name }: CartProductsProps) => {
  const { cart } = useCartContext()!;
  const [isLoading, products] = useCartProducts(cart);
  console.log(products);
  return (
    <section className="cards">
      {isLoading ? (
        <Loading />
      ) : (
        <>
          {products?.map((product) => (
            <CartCard key={product.id} product={product} />
          ))}
          <Total products={products} />
          <SecondaryButton>
            Place order<i className="bi bi-arrow-right"></i>
          </SecondaryButton>
        </>
      )}
    </section>
  );
};
