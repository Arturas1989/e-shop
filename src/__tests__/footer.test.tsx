import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Footer } from '../components';

describe('footer components renders', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});