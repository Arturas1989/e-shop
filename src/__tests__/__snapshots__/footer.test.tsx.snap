// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer components renders renders correctly 1`] = `
<footer
  id="footer"
>
  <div
    className="footer-bar"
  >
    <div
      className="copyright"
    >
      Â© 2024 e-shop. All rights reserved
    </div>
    <div
      className="socials"
    >
      <a
        href="/"
        rel="noreferer"
        target="_blank"
      >
        <i
          className="bi bi-instagram"
        />
      </a>
      <a
        href="/"
        rel="noreferer"
        target="_blank"
      >
        <i
          className="bi bi-twitter"
        />
      </a>
      <a
        href="/"
        rel="noreferer"
        target="_blank"
      >
        <i
          className="bi bi-github"
        />
      </a>
    </div>
  </div>
</footer>
`;
